.github/workflows/on-pull-request.yaml

```diff
@@ -38,5 +38,14 @@ jobs:
           ./scripts/validatei18n_files.sh
         shell: bash
 
-      - name: Build code (includes tests)
+      - name: Check for circular dependencies
```

Comment:
Why not use the shortcut `yarn test` that does all checks ? So if we add a new check, it will likely be added there vs. having to add it to this list.

Since we are at it, I think the name `test` is not appropriate anymore, maybe rename it to something like `validate` or `run-checks` ...

Reply:
Absolutely agree with the semantics of `test`... this is not testing anything!
However, I don't agree with having an all-in-one command, the main reason is because it makes the job log harder to read.
By separating these steps you can easily find at  what step there is a failure and fix it swiftly.

Reply:
sure